Parsing body with `all: true` broken in hono 4.3.3 and above

### What version of Hono are you using?

4.3.5

### What runtime/platform is your app running on?

Cloudflare Workers

### What steps can reproduce the bug?

Parsing the request body with `all` set to `true` (`c.req.parseBody({ all: true })`) is broken in Hono version 4.3.3 and above (tested with 4.3.3, 4.3.4, 4.3.5).

Code that works in 4.3.2:

```typescript
app.post('/upload', async (c) => {
  const body = await c.req.parseBody<{ file: File | File[] }>({ all: true })
  const files = Array.isArray(body['file']) ? body['file'] : [body['file']]

  if (files.length === 0) {
    c.status(400)
    return c.json({ error: 'No files' })
  }

  return c.json({ fileNames: files.map((file) => file.name) })
})
```

This can be tested with the following `curl` command:

```shell
curl -F file=@test1.txt -F file=@test2.txt http://localhost:8787/upload
```

### What is the expected behavior?

Parsing all form values into the response, this works in Hono 4.3.2 and below.

### What do you see instead?

Running in local development results in the error below: 

```
> wrangler dev src/index.ts

 ⛅️ wrangler 3.55.0
-------------------
⎔ Starting local server...
[wrangler:inf] Ready on http://localhost:8787
✘ [ERROR] TypeError: formKey is not iterable

      at handleParsingAllValues
  (file:///Users/paul/tmp/hono-test/node_modules/.pnpm/hono@4.3.5/node_modules/hono/dist/utils/body.js:35:27)
      at null.<anonymous>
  (file:///Users/paul/tmp/hono-test/node_modules/.pnpm/hono@4.3.5/node_modules/hono/dist/utils/body.js:25:7)
      at convertFormDataToBodyData
  (file:///Users/paul/tmp/hono-test/node_modules/.pnpm/hono@4.3.5/node_modules/hono/dist/utils/body.js:20:12)
      at parseFormData
  (file:///Users/paul/tmp/hono-test/node_modules/.pnpm/hono@4.3.5/node_modules/hono/dist/utils/body.js:14:12)
      at async HonoRequest.parseBody
  (file:///Users/paul/tmp/hono-test/node_modules/.pnpm/hono@4.3.5/node_modules/hono/dist/request.js:59:24)
      at async Array.<anonymous> (file:///Users/paul/tmp/hono-test/src/index.ts:6:16)
      at async jsonError
  (file:///Users/paul/tmp/hono-test/node_modules/.pnpm/wrangler@3.55.0_@cloudflare+workers-types@4.20240502.0/node_modules/wrangler/templates/middleware/middleware-miniflare3-json-error.ts:22:10)
      at async drainBody
  (file:///Users/paul/tmp/hono-test/node_modules/.pnpm/wrangler@3.55.0_@cloudflare+workers-types@4.20240502.0/node_modules/wrangler/templates/middleware/middleware-ensure-req-body-drained.ts:5:10)


[wrangler:inf] POST /upload 500 Internal Server Error (17ms)
```

### Additional information

_No response_

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -11,6 +11,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
       - uses: oven-sh/setup-bun@v1
         with:
           bun-version: '1.0.25'

--- a/deno_dist/utils/body.ts
+++ b/deno_dist/utils/body.ts
@@ -70,14 +70,12 @@ function convertFormDataToBodyData<T extends BodyData = BodyData>(
 }
 
 const handleParsingAllValues = (form: BodyData, key: string, value: FormDataEntryValue): void => {
-  const formKey = form[key] as (string | File)[]
+  const formKey = form[key]
 
-  if (form[key] && Array.isArray(form[key])) {
-    formKey.push(value)
-  } else if (form[key]) {
-    const parsedKey = [...formKey].join('').replace(',', '')
-
-    form[key] = [parsedKey, value]
+  if (formKey && Array.isArray(formKey)) {
+    ;(form[key] as (string | File)[]).push(value)
+  } else if (formKey) {
+    form[key] = [formKey, value]
   } else {
     form[key] = value
   }

--- a/src/utils/body.test.ts
+++ b/src/utils/body.test.ts
@@ -64,6 +64,24 @@ describe('Parse Body Util', () => {
     })
   })
 
+  it('should parse multiple values if values are `File`', async () => {
+    const file1 = new File(['foo'], 'file1', {
+      type: 'application/octet-stream',
+    })
+    const file2 = new File(['bar'], 'file2', {
+      type: 'application/octet-stream',
+    })
+    const data = new FormData()
+    data.append('file', file1)
+    data.append('file', file2)
+
+    const req = createRequest(FORM_URL, 'POST', data)
+
+    expect(await parseBody(req, { all: true })).toEqual({
+      file: [file1, file2],
+    })
+  })
+
   it('should parse multiple values if key ends with `[]`', async () => {
     const data = new FormData()
     data.append('file[]', 'aaa')

--- a/src/utils/body.ts
+++ b/src/utils/body.ts
@@ -70,14 +70,12 @@ function convertFormDataToBodyData<T extends BodyData = BodyData>(
 }
 
 const handleParsingAllValues = (form: BodyData, key: string, value: FormDataEntryValue): void => {
-  const formKey = form[key] as (string | File)[]
+  const formKey = form[key]
 
-  if (form[key] && Array.isArray(form[key])) {
-    formKey.push(value)
-  } else if (form[key]) {
-    const parsedKey = [...formKey].join('').replace(',', '')
-
-    form[key] = [parsedKey, value]
+  if (formKey && Array.isArray(formKey)) {
+    ;(form[key] as (string | File)[]).push(value)
+  } else if (formKey) {
+    form[key] = [formKey, value]
   } else {
     form[key] = value
   }

